client/question/components/SpeechBalloonDirectives.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for displaying speech balloons.
+ */
+
+tie.directive('tieSpeechBalloonContainer', ['$timeout', function($timeout) {
+  return {
+    restrict: 'A',
+    link: function(scope, iElement) {
+      if (iElement[0].nodeName !== "DIV") {
```

Comment:
- Why is this check needed?
- What does the "i" stand for in iElement?

Reply:
If I change restrict property above to E, this is no longer relevant. i stands for instance, but this can be anything (let me know if you prefer a different name).